### ISSUE TITLE
Enum.GetName vs FastEnum.GetName not found inconsistency

### ISSUE THREAD

**Issue by LukaszRozmej** [xin9le/FastEnum#11]
When `Enum.GetName` doesn't find a value it returns a `null`, `FastEnum.GetName` throws `KeyNotFoundException`. `FastEnum.GetName` should either return `null` or `FastEnum.TryGetName` method should be added.

**Comment by xin9le** [comment]
Thank you for telling me inconsistency. Following is a simple reproduction code using LINQPad. Since this behavior is a missing consideration, I'll consider how it should be treated.

```cs
void Main()
{
    Enum.GetName(Fruits.Banana).Dump("Enum.GetName(Fruits.Banana)");
    Enum.GetName<Fruits>(default).Dump("Enum.GetName<Fruits>(default)");

    FastEnum.GetName(Fruits.Banana).Dump("FastEnum.GetName(Fruits.Banana)");
    FastEnum.GetName<Fruits>(default).Dump("FastEnum.GetName<Fruits>(default)");
}

public enum Fruits
{
    Banana = 1,
}
```

**Comment by xin9le** [comment]
I'm very sorry for this workaround, and I currently consider returning `null` even if accepting breaking changes.

https://github.com/NethermindEth/nethermind/commit/d9267b7984b12253e2ced75d63996eb5f1f4fb6d#diff-56b7dbcc008eae2e1f2b1d5f56bb8b6d1de62655f43a408bcfe0013da38beb59R194-R196

**Comment by LukaszRozmej** [comment]
Yes workaround is not hard, though probably affects performance a bit and breaks a bit the idea behind this library to be in-plave replacement of original Enum class.

**Comment by xin9le** [comment]
I have just released the new package that version is *1.8.0*. The `.GetName()` now returns `null` when called with value that isn't defined in enum type. A workaround using the `.IsDefined()` isn't required. Please try it out.
